core/templates/dev/head/pages/exploration_player/ImagePreloaderService.js

```diff
@@ -52,11 +52,26 @@ oppia.factory('ImagePreloaderService', [
       AssetsBackendApiService.loadImage(
         ExplorationContextService.getExplorationId(), filename)
         .then(function(loadedImageFile) {
+          if (_isInFailedDownload(loadedImageFile.filename)) {
+            _removeFromFailedDownload(loadedImageFile.filename);
+          }
           var objectUrl = URL.createObjectURL(loadedImageFile.data);
           onLoadCallback(objectUrl);
         });
     };
 
+    /**
+     * Removes the given filename from the _failedDownload.
+     * @param {string} filename - The filename of the file which is to be
+     *                            removed from the _failedDownload array.
+     */
+    var _removeFromFailedDownload = function(filename) {
+      var index = _failedDownload.indexOf(loadedImageFile.filename);
+      if (index >= 0) {
```

Comment:
We don't need to check this, because we already check this in `_isInFailedDownload`.

Reply:
Done!